### PR TITLE
Add ORAS Artifact Reference Types docs

### PR DESCRIPTION
**Pull request by SteveLasker**
Add docs for how to use preview ORAS builds to push and discover ORAS Artifacts

Signed-off-by: Steve Lasker <stevenlasker@hotmail.com>

### CODE REVIEW

File: docs/CLI/6_reference_types.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+The focus on distributing secure supply chain artifacts has driven new innovations for supporting signatures, software bill of materials (SBoMs) and security scan results as artifacts that may be pushed, discovered and pulled, based on the tag or digest reference of a subject artifact.
+
+See the [ORAS Artifacts Spec scenarios][oras-artifacts-scenarios] for more details.
+
+## ORAS Artifact Spec Support
+
+Install the latest alpha build of the oras cli at: https://github.com/oras-project/oras/releases/tag/v0.2.1-alpha.1
+
+## Configure An Oras Artifact Enabled Registry
```
Comment (sajayantony) [inline docs/CLI/6_reference_types.md:13]: Need to uppercase `ORAS`

File: docs/CLI/6_reference_types.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+See [Registry Support](#registry-support)
+
+## Pushing Reference Types
+
+The following walkthrough will generate the following graph of artifacts:
```
Comment (sajayantony) [inline docs/CLI/6_reference_types.md:19]: Reword the double following 
The following walkthrough will generate the graph of artifacts shown below.

File: docs/CLI/6_reference_types.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  ```
+
+- Create a sample signature to the container image
+
+  ```bash
+  echo '{"artifact": "'${IMAGE}'", "signature": "pat hancock"}' > signature.json
+  ```
+
+- Push the signature to the registry, as a reference to the $IMAGE:
+
```
Comment (sajayantony) [inline docs/CLI/6_reference_types.md:47]: container image. Would avoid using $IMAGE as its a shell variable.

File: docs/CLI/6_reference_types.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  ```bash
+  oras push $REGISTRY/$REPO \
+      --artifact-type 'signature/example' \
+      --subject $IMAGE \
+      ./signature.json:application/json
+  ```
+
+## Discovering Artifact References
+
+The ORAS Artifacts Specification defines a [referrers API][oras-artifacts-referrers] for discovering references to a `subject` artifact. In the above case, oras discover can show the the list of references to the $IMAGE
```
Comment (sajayantony) [inline docs/CLI/6_reference_types.md:57]: container image.

File: docs/CLI/6_reference_types.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+- Using `oras discover`, view the graph of artifacts now stored in the registry
+
+  ```bash
+  oras discover -o tree $IMAGE
+  ```
+- The output shows the beginning of a graph of artifacts, where the signature is viewed as a child of the $IMAGE
```
Comment (sajayantony) [inline docs/CLI/6_reference_types.md:64]: container image.

File: docs/CLI/6_reference_types.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+- Create and push a sample Software Bill of Materials to the registry:
+  ```bash
+  echo '{"version": "0.0.0.0", "artifact": "'${IMAGE}'", "contents": "good"}' > sbom.json
+
+  oras push $REGISTRY/$REPO \
+    --artifact-type 'sbom/example' \
+    --subject $IMAGE \
+    ./sbom.json:application/json
+  ```
+- Sign the SBoM
```
Comment (sajayantony) [inline docs/CLI/6_reference_types.md:85]: Try using the linter in VSCode . Code fences needs a blank line following and preceding it.

File: docs/CLI/6_reference_types.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+- See [ORAS Artifacts Community](https://github.com/oras-project/artifacts-spec#community) for how to get engaged.
+
+### CNCF Distribution with ORAS Artifacts Support
+
+A reference implementation of the ORAS Artifacts Spec is available at [github.com/oras-project/distribution](https://github.com/oras-project/distribution)
+
+To run distribution locally:
+  ```bash
+  docker run -d -p 5000:5000 ghcr.io/oras-project/registry:v0.0.3-alpha
+  REGISTRY=localhost:5000
```
Comment (sajayantony) [inline docs/CLI/6_reference_types.md:143]: Why do you need this environment variable to run the registry locally?

**Review by sajayantony — CHANGES_REQUESTED** [review]

File: docs/CLI/6_reference_types.md (hunk 6 above)
Comment (SteveLasker) [reply docs/CLI/6_reference_types.md:85]: Which linter are you using? I'm not seeing warnings around indented bullets.

File: docs/CLI/6_reference_types.md (hunk 7 above)
Comment (SteveLasker) [reply docs/CLI/6_reference_types.md:143]: It's used by several of the commands in the script. This section sets up the variables to CNCF Distribution, ACR, and eventually ECR, Docker Hub, others

**Review by sajayantony — APPROVED** [review]

**Pull request merged**